packages/gatsby/src/schema/data-tree-utils.js

```diff
@@ -18,6 +18,8 @@ const areAllSameType = list =>
 const isEmptyObjectOrArray = (obj: any): boolean => {
   if (obj === INVALID_VALUE) {
     return true
+  } else if (_.isDate(obj)) {
+    return false
```

Comment:
does the below branch not catch this already?

Reply:
No it's reported as an empty object

Reply:
weeiiiird